src/pybind/mgr/influx/module.py

```diff
@@ -265,70 +342,74 @@ def send_to_influx(self):
-                               self.config['hostname'], self.config['port'])
+            databases = self.client.get_list_database()
+            if {'name': database} not in databases:
+                self.log.info("Database '%s' not found, trying to create "
+                              "(requires admin privs). You can also create "
+                              "manually and grant write privs to user "
+                              "'%s'", self.config['database'],
+                              self.config['username'])
+                self.client.create_database(database)
+                self.client.create_retention_policy(name='8_weeks',
+                                                    duration='8w',
+                                                    replication='1',
+                                                    default=True,
+                                                    database=database)
+        except (RequestException, InfluxDBClientError) as e:
             self.set_health_checks({
-                'MGR_INFLUX_SEND_FAILED': {
+                'MGR_INFLUX_DB_LIST_FAILED': {
                     'severity': 'warning',
-                    'summary': 'Failed to send data to InfluxDB server at %s:%d'
-                               ' due to an connection error'
-                               % (self.config['hostname'], self.config['port']),
+                    'summary': 'Failed to list/create InfluxDB database',
                     'detail': [str(e)]
                 }
             })
-        except InfluxDBClientError as e:
-            if e.code == 404:
-                self.log.info("Database '%s' not found, trying to create "
-                              "(requires admin privs).  You can also create "
-                              "manually and grant write privs to user "
-                              "'%s'", self.config['database'],
-                              self.config['username'])
-                client.create_database(self.config['database'])
-                client.create_retention_policy(name='8_weeks', duration='8w',
-                                               replication='1', default=True,
-                                               database=self.config['database'])
-            else:
-                self.set_health_checks({
-                    'MGR_INFLUX_SEND_FAILED': {
-                        'severity': 'warning',
-                        'summary': 'Failed to send data to InfluxDB',
-                        'detail': [str(e)]
-                    }
-                })
-                raise
+            self.log.exception('Failed to list/create InfluxDB database')
+            return False
+
+        self.log.debug('Gathering statistics')
+        points = self.gather_statistics()
+        for chunk in self.chunk(points, self.config['batch_size']):
+            self.queue.put(chunk)
+
+        self.log.debug('Queue currently contains %d items', self.queue.qsize())
```

Comment:
It would be good to have a bound on the queue length in case something unexpected goes wrong with the sender threads.  Could count how many chunks we enqueue in one pass through this function, and then check if the queue is e.g. more than 10x that number of chunks (and set some kind of flag to stop enqueuing things if so)

Reply:
Queue was already set to 100 items at max